cost-analyzer/templates/query-service-deployment-template.yaml

```diff
@@ -0,0 +1,113 @@
+{{- if and (not .Values.agent) (not .Values.cloudAgent) (.Values.kubecostDeployment) }}
+{{- if and (.Values.kubecostDeployment.queryServiceReplicas) (.Values.kubecostDeployment.queryServiceReplicas.enabled) }}
+apiVersion: apps/v1
+kind: StatefulSet
```

Comment:
Why is query service a StatefulSet? I apologize for my lack of context. If it is actually supposed to be/needs to be a StatefulSet, we should probably change the file name from `deployment-template` to `statefulset-template`.

Reply:
This is a conversation I had with Ajay, we're using a StatefulSet so that each replica can have its own PV. And yes, the template name should definitely say statefulset, not deployment.

Reply:
I saw you changed the PV mounts on the StatefulSet to both be read only (because I asked about it). If all PV mounts are read only, do we need a StatefulSet? Perhaps this is the question point; Ajay may have insight or knowledge here about PVs/configs of cost-model that I'm forgetting.

Reply:
Just ran some tests, I was wrong to set them to readOnly to begin with. I've updated that configuration, none of them are readOnly now.